common/sgx/quote.c

```diff
@@ -529,6 +547,220 @@ oe_result_t oe_verify_quote_with_sgx_endorsements(
+                NULL);
+        }
+
+        // Call internal API to verify QvE identity
+        OE_CHECK_MSG(
+            oe_verify_qve_report_and_identity(
+                quote,
+                (uint32_t)quote_size,
+                &qve_report_info,
+                expiration_check_date,
+                collateral_expiration_status,
+                quote_verification_result,
+                supplemental_data,
+                supplemental_data_size_out,
+                qve_isvsvn_threshold),
+            "Failed to check QvE report and identity",
+            oe_result_str(result));
+
+        result = OE_OK;
+    }
+
+#else
+
+    result = sgx_verify_quote(
+        &_ecdsa_uuid,
+        NULL,
+        0,
+        quote,
+        (uint32_t)quote_size,
+        expiration_check_date,
+        &collateral_expiration_status,
+        &quote_verification_result,
+        NULL,
+        0,
+        supplemental_data,
+        MAX_SUPPLEMENTAL_DATA_SIZE,
+        &supplemental_data_size_out,
+        *(uint32_t*)(sgx_endorsements->items[OE_SGX_ENDORSEMENT_FIELD_VERSION]
+                         .data),
+        (void*)(sgx_endorsements->items[OE_SGX_ENDORSEMENT_FIELD_TCB_INFO]
+                    .data),
+        sgx_endorsements->items[OE_SGX_ENDORSEMENT_FIELD_TCB_INFO].size,
+        (void*)(sgx_endorsements
+                    ->items[OE_SGX_ENDORSEMENT_FIELD_TCB_ISSUER_CHAIN]
+                    .data),
+        sgx_endorsements->items[OE_SGX_ENDORSEMENT_FIELD_TCB_ISSUER_CHAIN].size,
+        (void*)(sgx_endorsements->items[OE_SGX_ENDORSEMENT_FIELD_CRL_PCK_CERT]
+                    .data),
+        sgx_endorsements->items[OE_SGX_ENDORSEMENT_FIELD_CRL_PCK_CERT].size,
+        (void*)(sgx_endorsements
+                    ->items[OE_SGX_ENDORSEMENT_FIELD_CRL_PCK_PROC_CA]
+                    .data),
+        sgx_endorsements->items[OE_SGX_ENDORSEMENT_FIELD_CRL_PCK_PROC_CA].size,
+        (void*)(sgx_endorsements
+                    ->items[OE_SGX_ENDORSEMENT_FIELD_CRL_ISSUER_CHAIN_PCK_CERT]
+                    .data),
+        sgx_endorsements
+            ->items[OE_SGX_ENDORSEMENT_FIELD_CRL_ISSUER_CHAIN_PCK_CERT]
+            .size,
+        (void*)(sgx_endorsements->items[OE_SGX_ENDORSEMENT_FIELD_QE_ID_INFO]
+                    .data),
+        sgx_endorsements->items[OE_SGX_ENDORSEMENT_FIELD_QE_ID_INFO].size,
+        (void*)(sgx_endorsements
+                    ->items[OE_SGX_ENDORSEMENT_FIELD_QE_ID_ISSUER_CHAIN]
+                    .data),
+        sgx_endorsements->items[OE_SGX_ENDORSEMENT_FIELD_QE_ID_ISSUER_CHAIN]
+            .size);
+
+    if (result != OE_PLATFORM_ERROR)
+    {
+        OE_TRACE_INFO(
+            "SGX DCAP QVL-based quote verification is used, res: %s\n", result);
+
+        if (result != OE_OK)
+        {
+            OE_RAISE_MSG(
+                result,
+                "SGX QVL-based quote verification failed with error 0x%x",
+                result);
+        }
+
```

Comment:
If both ECDSA quote verification and QvE Identity verification passed, then I think it's ok to set `result = OE_OK`.  Any failure would cause `goto end` before setting result to OE_OK. 
Otherwise, it should be a bug.